inference-engine/tests/ngraph_functions/src/lstm_cell.cpp

```diff
@@ -27,9 +28,20 @@ std::shared_ptr<ngraph::Node> makeLSTM(const std::vector<ngraph::Output<Node>>&
         return std::make_shared<ngraph::opset4::LSTMCell>(in[0], in[1], in[2], W, R, B, hidden_size, activations,
                                                           activations_alpha, activations_beta, clip);
     } else {
-        std::vector<float> lenghts(in[0].get_shape()[0], in[0].get_shape()[1]);
-        auto seq_lenghts = ngraph::builder::makeConstant(in[0].get_element_type(), constants[3], lenghts, false);
-        return std::make_shared<ngraph::opset5::LSTMSequence>(in[0], in[1], in[2], seq_lenghts, W, R, B, hidden_size, direction,
+        std::shared_ptr<Node> seq_lengths;
+        if (mode == ngraph::helpers::SequenceTestsMode::PURE_SEQ
+            || mode == ngraph::helpers::SequenceTestsMode::CONVERT_TO_TI_MAX_SEQ_LEN_CONST) {
+            std::vector<float> lengths(in[0].get_shape()[0], in[0].get_shape()[1]);
+            seq_lengths = ngraph::builder::makeConstant(element::i64, constants[3], lengths, false);
+        } else if (mode == ngraph::helpers::SequenceTestsMode::CONVERT_TO_TI_RAND_SEQ_LEN_CONST) {
+            for (size_t  i = 0; i <= in[0].get_shape().at(0); ++i) {
+                std::vector<float> lengths;
+                seq_lengths = ngraph::builder::makeConstant(element::i64, constants[3], lengths, true, in[0].get_shape()[1], 0);
+            }
+        } else {
+            seq_lengths = in.at(3).get_node_shared_ptr();
```

Comment:
`at` throws an exception if element doesn't exist. Are we sure that `in` vector has 4 elements.

Reply:
else branch is only for modes == CONVERT_TO_TI_RAND_SEQ_LEN_PARAM, CONVERT_TO_TI_MAX_SEQ_LEN_PARAM . So it means Seq_len input should be as a Parameter here. But maybe it's better to write an exception message instead of .at exception

Reply:
fixed